### PR TITLE
Introducing GenAI RAD Lab module

### PR DESCRIPTION
**Pull request by guptamukul-google**

### CODE REVIEW

**Review by acolver — CHANGES_REQUESTED** [review]
A general suggestion: Can you sort the TF Vars? First by group# then by order#

File: modules/gen_ai/variables.tf [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+  description = "List of groups that should be added as the owner of the created project. {{UIMeta group=1 order=6 updatesafe }}"
+  type        = list(string)
+  default     = []
+}
+
+variable "owner_users" {
+  description = "List of users that should be added as owner to the created project. {{UIMeta group=1 order=7 updatesafe }}"
+  type        = list(string)
+  default     = []
+}
```
[the remark this answers is not in the document]
Comment (guptamukul-google) [reply modules/gen_ai/variables.tf:226]: As per my understanding, RAD Lab UI automatically add Owner role to the user who requested the Module creation. Also we would like the user creating the module add other users / groups as Owner.

File: modules/gen_ai/variables.tf [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+variable "ip_cidr_range" {
+  description = "Unique IP CIDR Range for AI Notebooks subnet. {{UIMeta group=2 order=5 }}"
+  type        = string
+  default     = "10.142.190.0/24"
+}
+
+variable "machine_type" {
+  description = "Type of VM you would like to spin up. {{UIMeta group=3 order=6 }}"
```
[the remark this answers is not in the document]
Comment (guptamukul-google) [reply modules/gen_ai/variables.tf:193]: Exactly! It is difficult to keep a region - machine type mapping therefore not adding machine types in option in UIMeta

File: modules/gen_ai/variables.tf [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+}
+
+variable "enable_services" {
+  description = "Enable the necessary APIs on the project.  When using an existing project, this can be set to false. {{UIMeta group=1 order=3 }}"
+  type        = bool
+  default     = true
+}
+
+variable "folder_id" {
+  description = "Folder ID where the project should be created. It can be skipped if already setting organization_id. Leave blank if the project should be created directly underneath the Organization node. {{UIMeta group=0 order=2 updatesafe }}"
```
[the remark this answers is not in the document]
Comment (guptamukul-google) [reply modules/gen_ai/variables.tf:157]: Yes `folder_id ` is `updatesafe` as there won't be any impact on project level and will just move the project from Org to Folder or from Folder to Folder.

File: modules/gen_ai/README.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+  create_project       = false
+  project_id_prefix    = "gen-ai-project-id"
+  enable_services      = false
+  
+  set_external_ip_policy          = false
+  set_shielded_vm_policy          = false
+  set_trustedimage_project_policy = false
+}
+````
```
[the remark this answers is not in the document]
Comment (acolver) [reply modules/gen_ai/README.md:99]: Looks like it still renders properly.

[the remark this answers is not in the document]
Comment (guptamukul-google) [reply modules/gen_ai/README.md:99]: Yes. 4 backticks works too.

**Comment by guptamukul-google** [comment]
UIMeta Order is the logical order for RAD Lab UI only but the variables in variables.tf are in alphabetical order for better readability of terraform variables and READMEs which are also built in the same order. 

Also its standardized across all modules.

**Review by acolver — APPROVED** [review]

**Pull request merged**